patroni/ha.py

```diff
@@ -78,12 +78,16 @@ def touch_member(self):
         self.dcs.touch_member(json.dumps(data, separators=(',', ':')))
 
     def clone(self, clone_member=None, msg='(without leader)'):
-        if self.state_handler.clone(clone_member):
-            logger.info('bootstrapped %s', msg)
-            cluster = self.dcs.get_cluster()
-            node_to_follow = self._get_node_to_follow(cluster)
-            self.state_handler.follow(node_to_follow, cluster.leader, True)
-        else:
+        try:
+            if self.state_handler.clone(clone_member):
+                logger.info('bootstrapped %s', msg)
+                cluster = self.dcs.get_cluster()
+                node_to_follow = self._get_node_to_follow(cluster)
+                self.state_handler.follow(node_to_follow, cluster.leader, True)
+            else:
+                logger.error('failed to bootstrap %s', msg)
+                self.state_handler.remove_data_directory()
+        except:
```

Comment:
Can you please elaborate why do you want to catch all exceptions here?
I see not a single reason to do that, because:
 1. All expected exceptions already handled underneath.
 2. Even if there is something really bad happening it will not disturb the main HA loop, because this method is executed only by `AsyncExecutor.run`, which at the end will catch all remaining things.